### PR TITLE
Consolidate config files

### PR DESCRIPTION
**Pull request by mattwthompson**
## Description
Removes 4 config files (5 with #307) and tidies some other stuff up

## Todos
 - [x] Do not install with setup.py
 - [x] Consolidate config files in pyproject.toml
 - [x] Use versioningit

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by j-wags** [comment]
The linting check is sad about something to do with versioningit, but I don't see any smoking guns in the file diff. It's complaining about using a shallow clone, but even though another workflow is explicitly doing a shallow clone, the linting action is unchanged and uses the default GH checkout workflow.

**Comment by mattwthompson** [comment]
I'd like to see that action dropped

File: devtools/conda-envs/psi4.yaml [hunk 1]
```diff
@@ -52,5 +54,6 @@ dependencies:
   - openff-units=0.2.1
   - postgresql
 
-  # Shim for QCFractal deployment issues
+  # Shims for QCFractal deployment issues
```
Comment (j-wags) [inline devtools/conda-envs/psi4.yaml:57]: (not blocking)
```suggestion
  # Shims for QCFractal deployment issues
  # see https://github.com/MolSSI/QCFractal/issues/864
  # Will be fixed in QCFractal > 0.56.0 once released
```

File: devtools/conda-envs/basic.yaml [hunk 2]
```diff
@@ -51,5 +52,6 @@ dependencies:
   - openmmforcefields
   - openff-fragmenter-base >=0.2.0
 
-  # Shim for QCFractal deployment issues
+  # Shims for QCFractal deployment issues
```
Comment (j-wags) [inline devtools/conda-envs/basic.yaml:55]:
```suggestion
  # Shims for QCFractal deployment issues
  # see https://github.com/MolSSI/QCFractal/issues/864
  # Will be fixed in QCFractal > 0.56.0 once released
```

**Review by j-wags — APPROVED** [review]
Thanks @mattwthompson. I saw this was marked as draft but it looked complete-ish to me. This is approved to to merge as-is, or after dropping the linting action.

File: devtools/conda-envs/psi4.yaml (hunk 1 above)
Comment (mattwthompson) [reply devtools/conda-envs/psi4.yaml:57]: I think these are fixes upstream and will try out removing them

**Pull request merged**